bentoctl/operator/registry.py

```diff
@@ -39,131 +43,152 @@ def list(self):
            our own operator or if cloning and tracking the operator in some other
            remote repository (other than github) then you can just pass the path
-           after the add command and it will register the local operator for you.
+           after the install command and it will register the local operator for you.
            This is a special case since the operator will not have an associated URL
            with it and hence cannot be updated using the tool.
 
         Args:
             name: Name of the operator
+            version: Version of the operator
         Returns:
             The the name of the operator installed.
         Raises:
             OperatorExists: There is another operator with the same name.
         """
-
-        if os.path.exists(name):
-            logger.info(f"adding operator from path ({name})")
-            content_path = name
-            git_url = None
-            git_branch = None
-
-        elif _is_official_operator(name) or _is_github_repo(name):
-            if _is_official_operator(name):
-                operator_repo = OFFICIAL_OPERATORS[name]
-            else:
-                operator_repo = name
-            owner, repo, branch = _fetch_github_info(operator_repo)
-            git_url = f"https://github.com/{owner}/{repo}.git"
-            git_branch = branch
-            content_path = _clone_git_repo(git_url, branch=branch)
-
-        elif _is_git_link(name):
-            git_url = name
-            content_path = _clone_git_repo(git_url)
-            git_branch = None
-
-        else:
-            OperatorNotAdded(
-                f"Operator not Added, Unable to parse {name}. "
-                "Please check docs to see the supported ways of adding operators."
+        if _is_official_operator(name):
+            if name in self.operators_list:
+                raise OperatorExists(operator_name=name)
+            operator_name, operator_info = self._install_official_operators(
+                name, version
             )
-
-        operator = Operator(content_path)
-
-        if operator.name in self.operators_list:
-            raise OperatorExists(operator_name=operator.name)
-
-        # move operator to bentoctl home
-        operator_path = _get_operator_dir_path(operator.name)
-        if git_url:  # if from git repo
-            shutil.copytree(content_path, operator_path)
-            is_local = False
-        else:  # if from local path
-            is_local = True
-            operator_path = content_path
-
-        # install operator dependencies
-        operator.install_dependencies()
-
-        operator.path = Path(operator_path)
-
-        self.operators_list[operator.name] = {
-            "path": os.path.abspath(operator.path),
-            "git_url": git_url,
-            "git_branch": git_branch,
-            "is_local": is_local,
-        }
+        else:
+            operator_name, operator_info = self._install_custom_operators(name)
```

Comment:
should we name this function to `_install_operator_from_path` to make it more explainable

Reply:
I think it is ok for now, since the only custom operator is from local path